src/main/io/gps.c

```diff
@@ -1540,9 +1994,47 @@ static bool UBLOX_parse_gps(void)
     uint32_t i;
 
     *gpsPacketLogChar = LOG_IGNORED;
+#define CLSMSG(cls, msg) (((cls) << 8) | (msg))
+    switch (CLSMSG(_class, _msg_id)) {
+    case CLSMSG(CLASS_MON, MSG_MON_VER): {
+        char *token = strtok(_buffer.ver.swVersion, " ");
+        while (token != NULL) {
+            float f = fastA2F(token);
+            if (f > 0.0f) {
+                int major = strtol(token, NULL, 10);
+                int minor = strtol(&token[major > 9 ? 3 : 2], NULL, 10);
+                gpsData.monVer.swVersion.firmwareVersion.major = major;
+                gpsData.monVer.swVersion.firmwareVersion.minor = minor;
+                break;
+            }
+            token = strtok(NULL, " ");
+        }
+        size_t j = 0;
+        while (j < 10) {
+            if (_buffer.ver.extension[(j * 30)] == '\0') {
+                break;
+            }
+            token = strtok(&_buffer.ver.extension[(j++ * 30)], "=");
+            while (token != NULL) {
+                if (strcmp(token, "PROTVER") == 0) {
+                    token = strtok(NULL, "=");
+                    int major = strtol(token, NULL, 10);
+                    int minor = strtol(&token[major > 9 ? 3 : 2], NULL, 10);
```

Comment:
```suggestion
                    char* p_next;
                    int major = strtol(token, &p_next, 10);
                    int minor = strtol(p_next + 1, NULL, 10);
```
It is not foolproof, but at least it will parse major > 100